File: unplugged_toaster/day03_two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        mp = {} 

        for i in range(0, len(nums)): 
            x = target - nums[i]

            if x in mp:
                return [mp[x], i]
        
            mp[nums[i]] = i

        return [] 

File: unplugged_toaster/day03_two-sum/two_sum.cpp
#include <iostream>
#include <vector>
#include <ranges>

class Solution { 
    public: 
        std::vector<int> static twoSum(std::vector<int>& nums, int target) { 

            for (int i = 0; i < nums.size(); i ++) {

                auto index = std::ranges::find(nums.begin() + i + 1, 
                nums.end(), target - nums[i]); 

                if (index != nums.end()) { 
                    return {i, 
                        static_cast<int>(std::distance(nums.begin(), index))};
                }
            }
            return {}; 
        }
};

int main() {
    std::vector<int> numbers = {3,4,5,6};
    auto retval = Solution::twoSum(numbers, 7); 

    for (auto& it: retval) { 
        std::cout << it << ","; 
    }
}